src/overlays/actors/ovl_En_Encount3/z_en_encount3.c

```diff
@@ -32,22 +34,174 @@ ActorInit En_Encount3_InitVars = {
+        new_var = this->unk16C * 0.52f;
+        if (new_var < this->actor.xzDistToPlayer) {
+            this->unk178 = ((this->unk16C * 0.03125f) * 0.001f) + 0.1f;
+        } else {
+            this->unk178 = 0.06f;
+        }
+
+        if (((this->unk16C + 50.0f) + BREG(0)) < this->actor.xzDistToPlayer) {
+            if (this->actionFunc == func_809AD194) {
+                this->unk148 = 0;
+                this->unk178 = 0.0f;
+                D_809AD810 = false;
+                if (((this->child != NULL) && (this->child->update != NULL)) && (this->child->colChkInfo.health > 0)) {
+                    Actor_Kill(this->child);
+                    this->child = NULL;
+                }
+                func_809AD058(this);
+            }
+        } else if (this->unk16C < this->actor.xzDistToPlayer) {
+            s16 i;
+
+            for (i = 0; i < PLAYER_BODYPART_MAX; i++) {
+                player->flameTimers[i] = Rand_S16Offset(0, 200);
+            }
+            player->isBurning = true;
+
+            sp3C = this->actor.world.pos.x - player->actor.world.pos.x;
+            sp38 = this->actor.world.pos.z - player->actor.world.pos.z;
+            if (!Play_InCsMode(play)) {
+                func_800B8D50(play, &this->actor, 10.0f, Math_Atan2S_XY(sp38, sp3C), 0.0f, 1);
+            }
+        }
+        this->child->colChkInfo = this->child->colChkInfo;
+    }
+
+    this->unk168 = this->unk16C;
+    this->unk168 /= 7666.0f;
+    if (this->actionFunc != func_809AD194) {
+        Math_ApproachZeroF(&this->unk170, 0.3f, 10.0f);
+        Math_ApproachZeroF(&this->unk160, 0.3f, 5.0f);
+        if (this->unk160 < 1.0f) {
+            play->unk_18880 = 0;
+        }
+    } else if (this->unk148 != 0) {
+        Math_ApproachF(&this->unk170, 255.0f, 0.4f, 10.0f);
+        Math_ApproachF(&this->unk160, 60.0f, 0.3f, 5.0f);
+    }
+    Math_ApproachF(&this->unk174, this->unk178, 0.3f, 0.03f);
+
+    this->unk164 = this->unk160 / 60.0f;
+    if (this->unk164 != 0.0f) {
+        play->envCtx.lightSettings.fogNear = (960 - play->envCtx.unk_C4.fogNear) * this->unk164;
+        play->envCtx.lightSettings.fogColor[0] = (40.0f - play->envCtx.unk_C4.fogColor[0]) * this->unk164;
+        play->envCtx.lightSettings.fogColor[1] = (10.0f - play->envCtx.unk_C4.fogColor[1]) * this->unk164;
+        play->envCtx.lightSettings.fogColor[2] = (0.0f - play->envCtx.unk_C4.fogColor[2]) * this->unk164;
+    }
+}
+
+void EnEncount3_Draw(Actor* thisx, PlayState* play) {
+    EnEncount3* this = THIS;
+    s32 pad;
+
+    if (this->unk170 > 0.0f) {
+        OPEN_DISPS(play->state.gfxCtx);
+
+        Gfx_SetupDL25_Xlu(play->state.gfxCtx);
+        Matrix_Push();
+
+        gDPPipeSync(POLY_XLU_DISP++);
+        gSPSegment(POLY_XLU_DISP++, 0x08,
+                   Gfx_TwoTexScroll(play->state.gfxCtx, 0, (s32)play->gameplayFrames, 0, 0x20, 0x40, 1,
+                                    (s32)play->gameplayFrames * -2, (s32)play->gameplayFrames * -8, 0x20, 0x20));
```

Comment:
```suggestion
                   Gfx_TwoTexScroll(play->state.gfxCtx, 0, play->gameplayFrames, 0, 0x20, 0x40, 1,
                                    play->gameplayFrames * -2, play->gameplayFrames * -8, 0x20, 0x20));
```
Sorry, just 1 small follow-up comment. Are these casts needed?

Reply:
yes